### PR TITLE
allow interval bounds with equal start and end

### PR DESCRIPTION
**Pull request by RNAer**
addressing #1441 . The demo for the edge cases are in the changes in `skbio/metadata/tests/test_intersection.py`. This is ready for review.

### CODE REVIEW

**Comment by RNAer** [comment]
@jairideout and @ebolyen , this is ready for you to review. just wanna make sure you guys don't miss it. :)

**Comment by jairideout** [comment]
@RNAer, apologies for the delay, one of us should have time to review on Monday.

File: skbio/metadata/_intersection.pyx [hunk 1]
```diff
@@ -186,7 +186,15 @@ cdef class IntervalNode:
         if self.cleft is not EmptyNode and self.cleft.maxend > start:
             self.cleft._intersect( start, end, results )
         # This interval
-        if ( self.end > start ) and ( self.start < end ):
+        if self.end == self.start:
+            send = self.end
+        else:
+            send = self.end - 1
```
Comment (ebolyen) [inline skbio/metadata/_intersection.pyx]: Add a comment indicating this is to mimic  the original behavior of `>` which should be enough to put a maintainer on the right track.

File: skbio/metadata/tests/test_intersection.py [hunk 2]
```diff
@@ -189,6 +189,19 @@ def test_find(self):
...
+        iv.insert(1, 1, 'foo')
+        iv.insert(3, 7, 'spam')
+        self.assertEqual(iv.find(0, 1), [])
+        self.assertEqual(iv.find(1, 1), ['foo'])
+        self.assertEqual(iv.find(1, 2), ['foo'])
+        self.assertEqual(iv.find(2, 3), [])
+        self.assertEqual(iv.find(3, 3), ['spam'])
+        self.assertEqual(iv.find(3, 4), ['spam'])
+        self.assertEqual(iv.find(6, 7), ['spam'])
+        self.assertEqual(iv.find(7, 7), [])
```
Comment (ebolyen) [inline skbio/metadata/tests/test_intersection.py]: Add a test that `iv.find(0, 8)` finds `['foo', 'spam']` just to make sure that "point ranges" interact as expected with normal ranges.

File: skbio/metadata/_intersection.pyx (hunk 1 above)
Comment (RNAer) [inline skbio/metadata/_intersection.pyx]: I don't get this. what comment you would like me to add here? Originally I didn't think about mimicing the old behaviour...

**Comment by RNAer** [comment]
@ebolyen , thanks for reviewing! updated according to your points except the comment you suggest adding.

File: skbio/metadata/_intersection.pyx (hunk 1 above)
Comment (ebolyen) [inline skbio/metadata/_intersection.pyx]: The original code was `self.end > start`, but when that changed to `self.end >= start` you needed to subtract one from `self.end` to match the new check (when start != end). Or at least I think that is why this code is written that way...

Comment (RNAer) [inline skbio/metadata/_intersection.pyx]: Actually nope. I tried other approaches. This is the most clear, working approach I can think of...

Comment (RNAer) [inline skbio/metadata/_intersection.pyx]: If you know the comment or other cleaner code, I would be happy to update with it. :)

Comment (ebolyen) [inline skbio/metadata/_intersection.pyx]: Ok let's not worry about this then.

**Comment by ebolyen** [comment]
@RNAer merging this broke master, tried rerunning this build to see what's up, but now it can't find the ref to pull the code...
